cotainr/container.py

```diff
@@ -22,8 +23,11 @@
 from tempfile import TemporaryDirectory
 
 from . import __version__ as _cotainr_version
+from . import tracing
 from . import util
 
+logger = logging.getLogger(__name__)
```

Comment:
may be a matter of style and different in python but should it not be ok just to have the static logger and alway call on that and not create a copy/ref here? - Not a blocker.

Reply:
This actually does provide a static (hierarchy of) logger(s). The logging module keeps track of all loggers added via `logging.getLogger` and returns a reference to a logger if it has already been created (which is also why we need to reload the logging modules in our tests, to reset this internal state). The pattern `logger = logging.getLogger(__name__)` is just the recommended way to maintain a hierarchy of loggers in line with the package/module hierarchy (see https://docs.python.org/3/library/logging.html#logger-objects for details).